crates/ra_ide/src/references/rename.rs

```diff
@@ -17,11 +17,9 @@ pub(crate) fn rename(
     position: FilePosition,
     new_name: &str,
 ) -> Option<RangeInfo<SourceChange>> {
-    let tokens = tokenize(new_name);
-    if tokens.len() != 1
-        || (tokens[0].kind != SyntaxKind::IDENT && tokens[0].kind != SyntaxKind::UNDERSCORE)
-    {
-        return None;
+    match single_token(new_name)?.token.kind {
```

Comment:
Hmm, I know that `Parse<T>` has `ok()` method. Maybe `ParsedToken` should declare the same?

https://github.com/rust-analyzer/rust-analyzer/blob/a108f22d835c8d67c93c66758d89371fff179b9a/crates/ra_syntax/src/lib.rs#L95-L102

Reply:
Well, since the tuple pattern has become a common practice here, lets leave it that way instead of `ParsedToken[s]`

Reply:
@matklad, `lex_single_token()` should return `Token` but not `SyntaxKind`, shouldn't it?
What about `try_single_token() -> Option<Token>` instead? So this would look like:
```suggestion
    match try_single_token(new_name)?.kind {
```

Reply:
Also, is it conventional to omit `get`, `parse`, `fetch` or other synonyms in function names?